tm2/pkg/sdk/abci.go

```diff
@@ -16,3 +16,11 @@ type BeginBlocker func(ctx Context, req abci.RequestBeginBlock) abci.ResponseBeg
 // Note: applications which set create_empty_blocks=false will not have regular block timing and should use
 // e.g. BFT timestamps rather than block height for any periodic EndBlock logic
 type EndBlocker func(ctx Context, req abci.RequestEndBlock) abci.ResponseEndBlock
+
+// BeginTxHook is a BaseApp-specific hook, called to modify the context with any
```

Comment:
This is not changing ABCI, which is in `bft/abci`, but the `BaseApp`.

Reply:
Leaving a comment here for historical reasons, since we discussed this on a call

There is something _seriously_ fishy about having to capture transaction processing events through hooks like this, and it's an indicator of a much bigger problem (that's waaaay out of the scope of this PR)

Reply:
Resolving this thread since it's fine for the interim, but just acknowledging it's new tech debt future us needs to tackle (famous words) :)